### DISCUSSION TITLE
Heterogeneous type curve models

### DISCUSSION THREAD

**Discussion by jcherw2** [GeoStat-Framework/welltestpy#30] [Ideas]
Is there any plan to include options for non-uniform aquifers such as channelized/strip- or multi radial-aquifer see eg.

https://journals.scholarsportal.info/details/00221694/v128i1-4/69_ptinatlsc.xml&sub=all   -channel strip

https://www.sciencedirect.com/science/article/abs/pii/002216948890025X  -radial composite

I used to have Fortran codes that calculated these curves

**Comment by MuellerSeb** [comment]
Hey there!

Thanks for your interest in welltestpy. These type-cruves basically would need to be added to [AnaFlow](https://github.com/GeoStat-Framework/AnaFlow) to be then used in welltestpy to fit them to given data.

The second link is in theory already implemented, since the solution of Buttler is a special case of the [extended generalized radial flow model](https://www.sciencedirect.com/science/article/pii/S0309170821001822?via%3Dihub) that we presented and use to calculate most type-curves in AnaFlow. See this [figure](https://www.sciencedirect.com/science/article/pii/S0309170821001822?via%3Dihub#fig0002).

There we would only need to add a new estimator to welltespy using this special case.

At the moment there a no plans to do this, but I will keep it in mind for the future. But contributions are welcome.

Cheers, Sebastian
